src/Symfony/Component/Serializer/Tests/SerializerTest.php

```diff
@@ -1279,6 +1280,36 @@ public function testNoCollectDenormalizationErrorsWithWrongEnumOnConstructor()
         }
     }
 
+    public function testGroupsOnClassSerialization()
+    {
+        $obj = new Fixtures\Attributes\GroupClassDummy();
+        $obj->setFoo('foo');
+        $obj->setBar('bar');
+
+        $obj2 = new Fixtures\Annotations\GroupClassDummy();
+        $obj2->setFoo('foo');
+        $obj2->setBar('bar');
```

Comment:
I would only test with the attributes fixture here and add another test to `Symfony\Component\Serializer\Tests\Mapping\Loader\AnnotationLoaderTestCase` to make sure annotations and attributes are loaded the same way.

Reply:
@derrabus done https://github.com/symfony/symfony/pull/49594/commits/ac82af881f790176c0c7d4e8f5206097f5f7db86